File: app/src/main/java/com/example/tree/zhihu/adapters/ColumnMessgeAdapter.java
package com.example.tree.zhihu.adapters;

import android.content.Context;
import android.content.Intent;
import android.graphics.Bitmap;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.example.tree.zhihu.activity.ColumnMessageActivity;
import com.example.tree.zhihu.gson.ColumnMessage;
import com.example.tree.zhihu.gson.NewsExtra;
import com.example.tree.zhihu.R;
import com.example.tree.zhihu.activity.WebContentActivity;

import java.util.List;

public class ColumnMessgeAdapter extends RecyclerView.Adapter<ColumnMessgeAdapter.ViewHolder> {
    Context context;
    ColumnMessage columnMessage;
    List<NewsExtra> newsExtras;
    List<Bitmap> bitmaps;


    public ColumnMessgeAdapter(Context context,ColumnMessage columnMessage,List<NewsExtra> newsExtras,List<Bitmap> bitmaps){
        this.context = context;
        this.columnMessage = columnMessage;
        this.bitmaps = bitmaps;
        this.newsExtras = newsExtras;
    }


    @NonNull
    @Override

    public ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        return new ViewHolder(LayoutInflater.from(context).inflate(R.layout.item_column_message,viewGroup,false));

    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder viewHolder, final int i) {
        viewHolder.column_extra_linear.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                ((ColumnMessageActivity)context).getViewPager().setCurrentItem(1);
                ((ColumnMessageActivity)context).SetIdearFragment(""+columnMessage.getStories().get(i).getId());
            }
        });
        viewHolder.te_title.setText(columnMessage.getStories().get(i).getTitle());
        viewHolder.te_date.setText(columnMessage.getStories().get(i).getDate());
        if (newsExtras.size() > i && bitmaps.size() > i){
            viewHolder.te_popular.setText(""+newsExtras.get(i).getPopularity());
            viewHolder.te_comment.setText(""+newsExtras.get(i).getComments());
            viewHolder.image_title.setImageBitmap(bitmaps.get(i));
        }
        viewHolder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(context, WebContentActivity.class);
                intent.putExtra("id", columnMessage.getStories().get(i).getId());
                context.startActivity(intent);
            }
        });

    }

    @Override
    public int getItemCount() {
        return columnMessage.getStories().size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        TextView te_title;
        TextView te_comment;
        TextView te_popular;
        TextView te_date;
        ImageView image_title;
        LinearLayout column_extra_linear;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            te_comment =(TextView) itemView.findViewById(R.id.text_agrees_with_the_column_message);
            te_popular = (TextView)itemView.findViewById(R.id.text_review_column_message);
            te_date =(TextView) itemView.findViewById(R.id.item_column_message_date);
            te_title = (TextView) itemView.findViewById((R.id.item_column_message_title));
            image_title = (ImageView) itemView.findViewById(R.id.item_column_message_picture_title);
            column_extra_linear = (LinearLayout) itemView.findViewById((R.id.column_extra));
        }
    }
}

File: app/src/main/java/com/example/tree/zhihu/adapters/ColumnMessgeAdapter.kt
package com.example.tree.zhihu.adapters

import android.content.Context
import android.content.Intent
import android.graphics.Bitmap
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView

import com.example.tree.zhihu.activity.ColumnMessageActivity
import com.example.tree.zhihu.gson.ColumnMessage
import com.example.tree.zhihu.gson.NewsExtra
import com.example.tree.zhihu.R
import com.example.tree.zhihu.activity.WebContentActivity
import kotlinx.android.synthetic.main.activity_column_message.*

class ColumnMessgeAdapter(internal var context: Context, internal var columnMessage: ColumnMessage, internal var newsExtras: List<NewsExtra>, internal var bitmaps: List<Bitmap>) : RecyclerView.Adapter<ColumnMessgeAdapter.ViewHolder>() {


    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): ViewHolder {
        return ViewHolder(LayoutInflater.from(context).inflate(R.layout.item_column_message, viewGroup, false))

    }

    override fun onBindViewHolder(viewHolder: ViewHolder, i: Int) {
        viewHolder.column_extra_linear.setOnClickListener {
            (context as ColumnMessageActivity).column_view_pager.currentItem = 1
            (context as ColumnMessageActivity).SetIdearFragment("" + columnMessage.stories[i].id)
        }
        viewHolder.te_title.text = columnMessage.stories[i].title
        viewHolder.te_date.text = columnMessage.stories[i].date
        if (newsExtras.size > i && bitmaps.size > i) {
            viewHolder.te_popular.text = "" + newsExtras[i].popularity
            viewHolder.te_comment.text = "" + newsExtras[i].comments
            viewHolder.image_title.setImageBitmap(bitmaps[i])
        }
        viewHolder.itemView.setOnClickListener {
            val intent = Intent(context, WebContentActivity::class.java)
            intent.putExtra("id", columnMessage.stories[i].id)
            context.startActivity(intent)
        }

    }

    override fun getItemCount(): Int {
        return columnMessage.stories.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var te_title: TextView
        var te_comment: TextView
        var te_popular: TextView
        var te_date: TextView
        var image_title: ImageView
        var column_extra_linear: LinearLayout

        init {
            te_comment = itemView.findViewById<View>(R.id.text_agrees_with_the_column_message) as TextView
            te_popular = itemView.findViewById<View>(R.id.text_review_column_message) as TextView
            te_date = itemView.findViewById<View>(R.id.item_column_message_date) as TextView
            te_title = itemView.findViewById<View>(R.id.item_column_message_title) as TextView
            image_title = itemView.findViewById<View>(R.id.item_column_message_picture_title) as ImageView
            column_extra_linear = itemView.findViewById<View>(R.id.column_extra) as LinearLayout
        }
    }
}
